### PR TITLE
[SYCL][NFC] Fix bugs in unit tests

### PR DESCRIPTION
**Pull request by againull**
* Currently PIImage object holds raw pointers to the data in std containers in the MBinaryDesc structure. If PIImage object gets moved or copied then these pointers get invalidated causing invalid memory access and flaky failures. Patch removes MBinaryDesc structure and changes the code to get those pointers on the fly. 
* Remove cend() iterator dereference in unit tests.
* Use different kernel identifiers for different tests.All device images are stored in the same program manager object and tests doesn't take that into account. So, using the same kernel identifier in different tests leads to multiple device images (related to different tests) in the program manager associated with the same kernel identifier. It leads to the situation when test's behaviour is affected by artifacts from another test.
* Fixed redefined compile/link/build plugin functions. Currently they may produce a result with redundant space symbol.

### CODE REVIEW

File: sycl/unittests/helpers/PiImage.hpp [hunk 1]
```diff
@@ -206,10 +206,90 @@ class PiImage {
...
+  /// Constructs a SYCL device image of the latest version.
+  PiImage(uint8_t Format, const std::string &DeviceTargetSpec,
+          const std::string &CompileOptions, const std::string &LinkOptions,
+          std::vector<unsigned char> Binary,
+          PiArray<PiOffloadEntry> OffloadEntries, PiPropertySet PropertySet)
+      : PiImage(PI_DEVICE_BINARY_VERSION, PI_DEVICE_BINARY_OFFLOAD_KIND_SYCL,
+                Format, DeviceTargetSpec, CompileOptions, LinkOptions, {},
+                std::move(Binary), std::move(OffloadEntries),
+                std::move(PropertySet)) {}
+
```
Comment (aelovikov-intel) [inline sycl/unittests/helpers/PiImage.hpp:221]: Why can't we use default for this one? Please add a comment.

**Review by aelovikov-intel — COMMENTED** [review]
I don't like this. Why can't we change the APIs so that `pi_device_binary_struct MBinaryDesc;` isn't a member of the class and generated on the fly instead?

File: sycl/unittests/helpers/PiImage.hpp (hunk 1 above)
Comment (againull) [reply sycl/unittests/helpers/PiImage.hpp:221]: This one was not added in this PR, it was here before: https://github.com/intel/llvm/pull/7070/files#diff-0e91a022ae2eb241758604e815c0f50613d9515ce8ef0ad5c865918f809cc91eL233-L245

**Comment by againull** [comment]
> I don't like this. Why can't we change the APIs so that `pi_device_binary_struct MBinaryDesc;` isn't a member of the class and generated on the fly instead?

Fixed to generate on the fly.

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**